harness/determined/experimental/unmanaged.py

```diff
@@ -151,102 +250,66 @@ def create_unmanaged_cluster_info(
     )
 
 
-def init(
-    *,
+def get_or_create_experiment_and_trial(
```

Comment:
i'm generally skeptical of `get_or_create`. is there a reason why we don't separate this method into separate `get`/`create`?

Reply:
what I've discovered while working on this is that separate get+create, especially from client side, will lead to conflicts or data races.

consider `ray/ray_hp_search.py` example — Ray Tune ASHA hp search. on my machine it'd spawn 16 independent trials simultaneously. let's say we do `get`, all 16 of them see 404 because the experiment does not exist yet, so they all do `create`, and here 1 of them will succeed and the other 15 will fail and we'll need to handle this (e.g. retry get again). this is not great.

instead, we could have all 16 do `PUT` request which does get-or-create atomically on the server.